src/Serializer.php

```diff
@@ -61,9 +79,20 @@ public function serialize(\Closure $closure)
      */
     public function unserialize($serialized)
     {
-        /** @var SerializableClosure $unserialized */
         $unserialized = unserialize($serialized);
 
+        if ($this->signingKey) {
+            list($data, $signature) = $unserialized;
+            if ($signature !== $this->calculateSignature($data)) {
```

Comment:
Also: this should use `hash_equals` since otherwise you'd leak the calculated signature (which could expose the key potentially)..